query.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/*
+ Copyright 2020 Matthias Bissinger (email: matthias.bissinger@fau.de)
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 2 of the License, or (at
+ your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+// define the API url
+function wp_nasaads_query_importer_url() {
+    return 'https://api.adsabs.harvard.edu/v1/';
+}
+
+
+// functions for handling query errors
+$wp_nasaads_query_importer_error = null;
+function wp_nasaads_query_importer_throw($msg, $type = 0) {
+    global $wp_nasaads_query_importer_error;
+    $wp_nasaads_query_importer_error = array('msg' => $msg, 'type' => $type);
+    return false;
+}
+function wp_nasaads_query_importer_get_error() {
+    global $wp_nasaads_query_importer_error;
+    return $wp_nasaads_query_importer_error;
+}
+
+
+// build the API query basend on the given attribute array
+function wp_nasaads_query_importer_build_query($atts, $fetch) {
+    $what = $atts['url'];
+    if (! is_null($what)) {
+        return $what;
+    }
+
+    # search query given explicitly
+    if (! is_null($atts['query'])) {
+        return $atts['query'];
+    }
+
+    # build search query from given fields
+    # (supported only if no library is requested)
+    $fields = array();
+    if (is_null($atts['library'])) {
+        if (! is_null($atts['author'])) {
+            # in case of "easy logic", i.e., there are no quotation marks,
+            # OR, ANDs, or parentheses, then we assume a single author
+            # and add quotation marks around her/his name
+            if (! preg_match('/["()]+|( AND )+|( OR )+/', $atts['author'])) {
```

Comment:
Can we allow AND or OR to be case insensitive? ADS API and website is case insensitive for these operands.